### ISSUE TITLE
Modular formatting

### ISSUE THREAD

**Issue by sebastiankuhnert** [chadparry/bibleref#8]
Is it possible to separate the text formatting from the "markup" formatting used for URLs?
One way to do this could be to define "outer styles" that receive both the parsed passage data structure and the formatted string as parameter. If more elementary control over the formatted chunks is needed, this could be achieved by enriching the passage/range data structure macros, using the respective formatted string as additional parameter.

Maybe this proposal becomes almost automatically realized, once the string formatting is done using bibleref's style system (issue #6): The latter could be statically used for text formatting, while bibleref-mouth's style system would remain in place for formatting of hyperlinks and the like.

**Comment by chadparry** [comment]
The biggest complication here is that the formatting and the target link are sometimes inextricably coupled. For example, see my comment in issue #9. Suppose that the reference is to "Mal. 3-4". If this is a link to biblegateway.com, then it gets formatted like so: `<a href="http://www.biblegateway.com/passage/?search=Malachi+3-4" >Mal. 3-4</a>`. On the other hand, a link to bibleserver.com would be formatted like so: `<a href="http://www.bibleserver.com/text/ESV/Malachi3" >Malachi 3</a>-<a href="http://www.bibleserver.com/text/ESV/Malachi4" >4</a>`. There could be other websites that have even different capabilities, (e.g. links must be to individual verses, not to ranges of verses). Because of all this, it may not be desirable to force authors to create a generic text format that can be used with any link target.

**Comment by sebastiankuhnert** [comment]
Yes, the different linking styles would need different parts to be linked. But having thought about it a bit further, each of the linking styles could use the text formatting macros provided by bibleref.sty (as proposed for the default style in issue #6). I think this can be done locally, just as the respective linking style needs. I'll have a look at both of these issues.
